### ISSUE TITLE
Your IMAP client does not appear to correctly implement Unicode messages, see https://tools.ietf.org/html/rfc6855.html

### ISSUE THREAD

**Issue by Popyllol** [svarshavchik/courier-libs#14]
Hi Sam,
it is funny to show Microsoft how bad they implement RFCs, but with your code starting at
https://github.com/svarshavchik/courier-libs/blob/ec5264047dfaf66c61cd0f4c9b8dd40463693722/imap/imapd.c#L797

you are annoing Microsoft users a little bit too much.

On every connect they get the message "Your IMAP client does not appear to correctly implement Unicode messages, see https://tools.ietf.org/html/rfc6855.html" which is not very kind.
I know you want the big bunch of shit called Microsoft to get rid of their ineptitude to write good code, but on the other side we have users who just want to use software like a spoon, knife or fork. Simple.

So it would be nice to have the opportunity to configure the annoyance level.

Thank you,
   Lars

**Comment by svarshavchik** [comment]
This only happens if you have a message that is either corrupted and its sending mail client fails to properly encode it, or it is an actual Unicode message. RFC 6855 now requires this message to be blocked from IMAP clients that do not implement RFC 6855.

I am tweaking the notifications and error reporting, trying to find some middle ground where the errors continue to be reported, but to a lesser degree. You can try the most recent development build and see how much of a difference it makes.

**Comment by lisergey** [comment]
it helped me to add UTF8=ACCEPT in IMAP_CAPABILITY 
it was first idea when I followed the recommended link in Outlook complaint.

so, maybe, it's worth to recommend it with the link altogether?

**Comment by svarshavchik** [comment]
imapd already advertises UTF8=ACCEPT, this does not need to be added to the capability setting.

**Comment by joshuaboniface** [comment]
I'm encountering the same issue, due to users with Outlook 2016. I'm using Courier from the Debian 10 stable repositories (`courier-imap` version 1.0.6). Reviewing the code nothing has changed in this regard since that release, so I hope I'm not missing something.

I've noticed two curious things:

1. Manually setting `UTF8=ACCEPT` *does* in fact make a difference, but only for the Inbox. Without the explicit capability, I was having users see this error message when they initially received the email, but with the setting they do not. This doesn't make much sense to me for exactly the reason you said - I see it twice in the pcaps, but it did do something.

2. The error does occur, though, when they delete an email or otherwise move it to another folder. I confirmed the findings in https://sourceforge.net/p/courier/mailman/courier-imap/thread/cone.1570792508.130442.7141.1004%40monster.email-scan.com/#msg36782872 that Outlook 2016 is indeed not appending `UTF8` to its `APPEND` command, which then triggers the error message.

I'm curious if you know of any additional workarounds we can try, short of "use a different mail client" which might not be possible?
